app/src/main/java/org/greenstand/android/TreeTracker/permission/Permissions.kt

```diff
@@ -0,0 +1,174 @@
+                                Text(text = stringResource(R.string.accept_location_permission_header)
+                                )
+                            },
+                            text = {
+                                Text(
+                                    text = stringResource(R.string.accept_location_permission_message),
+                                    color = Color.Green
+                                )
+                            },
+                            buttons = {
+                                Button(
+                                    modifier = Modifier.wrapContentSize(),
+                                    onClick = {
+                                        perm.launchPermissionRequest()
+                                        navController.popBackStack()
+                                    }
+                                ) {
+                                    Text(
+                                        text = stringResource(R.string.accept_permission)
+                                    )
+                                }
+                            })
+                    }
+
+                }
+            }
+            Manifest.permission.ACCESS_COARSE_LOCATION -> {
+                when {
+                    perm.hasPermission -> {
+                        // TODO Check if location is enabled from the LocationUpdateManager before calling enableLocation
+                        return
+                    }
+                    (perm.shouldShowRationale || !perm.permissionRequested) -> {
+                        AlertDialog(
+                            onDismissRequest = { navController.popBackStack() },
+                            title = {
+                                Text(text = stringResource(R.string.accept_location_permission_header))
+                            },
+                            text = {
+                                Text(
+                                    text = stringResource(R.string.accept_location_permission_message),
+                                    color = Color.Green
+                                )
+                            },
+                            buttons = {
+                                Button(
+                                    modifier = Modifier.wrapContentSize(),
+                                    onClick = {
+                                        perm.launchPermissionRequest()
+                                        navController.popBackStack()
+                                    }
+                                ) {
+                                    Text(
+                                        text = stringResource(R.string.accept_permission)
+                                    )
+                                }
+                            })
+                    }
+                }
+            }
+        }
+    }
+}
+
+fun isLocationEnabledCheck() {
+
+}

```

Comment:
This is not needed.

Reply:
This is actually needed but I need your help here. This function is to check if location/gps is enabled  which should return a boolean valueand it is only called after confirming permission has been accepted. My challenge is accessing fun isLocationEnabled() in the Locations.kt file(line 106) to return the value

Reply:
The isLocationEnabled method in the LocationUpdateManager is in the "model" layer of the app. There is the view/viewmodel/model layers. The model should only talk to the viewmodel layer. That means you can add the LocationUpdateManager to a viewmodel, then expose that locationEnabled through a state object.

This permission method doesn't have a viewmodel, but in this case it could make life easier if we make a viewmodel for this, since it's used in multiple screens. To do that, follow the convention the other viewmodels use.

Reply:
Great suggestion, I would try this out.